### PR TITLE
Rrefresh intermediary accounts test

### PR DESCRIPTION
**Pull request by antstalepresh**
Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: x/superfluid/keeper/stake_test.go [hunk 1]
```diff
@@ -663,334 +659,253 @@ func (suite *KeeperTestSuite) TestSuperfluidUnbondLock() {
...
-// 				// check unbonded amount is removed after refresh operation
-// 				refreshed := suite.App.BankKeeper.GetBalance(suite.Ctx, expAcc.GetAccAddress(), sdk.DefaultBondDenom)
-// 				suite.Require().True(refreshed.IsZero())
-// 			}
-// 		})
-// 	}
-// }
+			for index, intAccIndex := range tc.checkAccIndexes {
+				expAcc := intermediaryAccs[intAccIndex]
+				valAddr, err := sdk.ValAddressFromBech32(expAcc.ValAddr)
```
Comment (mattverse) [inline x/superfluid/keeper/stake_test.go:830]: Can we group two for loops together for code sanity? Something like this:
```
	for index, intAccIndex := range tc.checkAccIndexes {
				expAcc := intermediaryAccs[intAccIndex]
				multiplier, ok := multiplierByDenom[expAcc.Denom]
				if !ok {
					multiplier = originMultiplier
				}

				targetDelegation := intermediaryDels[index].Mul(multiplier).Quo(originMultiplier)
				lpTokenAmount := sdk.NewInt(1000000)
				decAmt := multiplier.Mul(lpTokenAmount.ToDec())
				asset := suite.App.SuperfluidKeeper.GetSuperfluidAsset(suite.Ctx, expAcc.Denom)
				targetAmount := suite.App.SuperfluidKeeper.GetRiskAdjustedOsmoValue(suite.Ctx, asset, decAmt.RoundInt())

				valAddr, err := sdk.ValAddressFromBech32(expAcc.ValAddr)
				suite.Require().NoError(err)

				delegation, found := suite.App.StakingKeeper.GetDelegation(suite.Ctx, expAcc.GetAccAddress(), valAddr)
				if targetAmount.IsPositive() {
					suite.Require().True(found)
					suite.Require().Equal(delegation.Shares, targetDelegation)
				} else {
					suite.Require().False(found)
				}
			}
```

**Review by mattverse — APPROVED** [review]
Minor reviews for code sanity, otherwise looks good! 
The maths to fix this test would have been painful haha

File: x/superfluid/keeper/stake_test.go (hunk 1 above)
Comment (antstalepresh) [reply x/superfluid/keeper/stake_test.go]: Merged two loops into one - thanks for finding this

File: x/superfluid/keeper/stake_test.go [hunk 2]
```diff
@@ -663,334 +659,244 @@ func (suite *KeeperTestSuite) TestSuperfluidUnbondLock() {
...
+			}
 
-// 			panic("e")
-// 		})
-// 	}
-// }
+			// refresh intermediary account delegations
+			suite.NotPanics(func() {
+				suite.App.SuperfluidKeeper.RefreshIntermediaryDelegationAmounts(suite.Ctx)
+			})
```
Comment (ValarDragon) [inline x/superfluid/keeper/stake_test.go:891]: How would this panic? Isn't the sensitive logic all wrapped in an `ApplyIfNoErr` function?

File: x/superfluid/keeper/stake_test.go [hunk 3]
```diff
@@ -663,334 +659,244 @@ func (suite *KeeperTestSuite) TestSuperfluidUnbondLock() {
...
+func (suite *KeeperTestSuite) TestRefreshIntermediaryDelegationAmounts() {
+	testCases := []struct {
+		name                string
+		validatorStats      []stakingtypes.BondStatus
+		delegatorNumber     int
+		superDelegations    []superfluidDelegation
+		roundOneMultipliers []osmoEquivalentMultiplier
+		roundTwoMultipliers []osmoEquivalentMultiplier
+		checkAccIndexes     []int64
+	}{
```
Comment (ValarDragon) [inline x/superfluid/keeper/stake_test.go:671]: I don't understand what this test is doing. It seems like it applies two refreshes for osmo equivalent TWAPs, and then at the end checks that every account index provided has 0 osmo in it?

How is this checking that the refresh operation applied as expected?

File: x/superfluid/keeper/stake_test.go (hunk 2 above)
Comment (antstalepresh) [reply x/superfluid/keeper/stake_test.go:891]: This won't panic - previously there was a case for panics - I will remove it

File: x/superfluid/keeper/stake_test.go (hunk 3 above)
Comment (antstalepresh) [reply x/superfluid/keeper/stake_test.go:671]: It checks if intermediary account does not have balance after refresh operation.
And it is checking if the operation is applied by checking the delegation amount change.

```
targetDelegation := intermediaryDels[index].Mul(multiplier).Quo(originMultiplier)
```

Here `targetDelegation` is expected delegation after refresh.

**Review by alexanderbez — APPROVED** [review]

File: x/superfluid/keeper/stake_test.go (hunk 3 above)
Comment (ValarDragon) [reply x/superfluid/keeper/stake_test.go:671]: Ohh I see, I very much misunderstood this on first read. Thanks for updating the name

**Pull request merged**